qiskit/passmanager/flow_controllers.py

```diff
@@ -10,149 +10,201 @@
+                    "options": options,
+                    alias: controllers.pop(alias),
+                }
+                instance = class_type(**init_args)
+
+        return instance
 
     @classmethod
-    def remove_flow_controller(cls, name):
-        """Removes a flow controller.
+    def add_flow_controller(
+        cls,
+        name: str,
+        controller: Type[BaseFlowController],
+    ):
+        """Adds a flow controller.
 
         Args:
-            name (string): Name of the controller to remove.
-        Raises:
-            KeyError: If the controller to remove was not registered.
+            name: Alias of controller class in the namespace.
+            controller: Flow controller class.
         """
-        if name not in cls.registered_controllers:
-            raise KeyError("Flow controller not found: %s" % name)
-        del cls.registered_controllers[name]
+        cls.registered_controllers[name] = controller
+        if name not in cls.hierarchy:
+            cls.hierarchy.append(name)
 
     @classmethod
-    def controller_factory(
+    def remove_flow_controller(
         cls,
-        passes: Sequence[GenericPass | "FlowController"],
-        options: dict,
-        **partial_controller,
+        name: str,
     ):
-        """Constructs a flow controller based on the partially evaluated controller arguments.
+        """Removes a flow controller.
 
         Args:
-            passes: passes to add to the flow controller.
-            options: PassManager options.
-            **partial_controller: Partially evaluated controller arguments in the form `{name:partial}`
+            name: Alias of the controller to remove.
 
         Raises:
-            PassManagerError: When partial_controller is not well-formed.
-
-        Returns:
-            FlowController: A FlowController instance.
+            KeyError: If the controller to remove was not registered.
         """
-        if None in partial_controller.values():
-            raise PassManagerError("The controller needs a condition.")
-
-        if partial_controller:
-            for registered_controller in cls.registered_controllers.keys():
-                if registered_controller in partial_controller:
-                    return cls.registered_controllers[registered_controller](
-                        passes, options, **partial_controller
-                    )
-            raise PassManagerError("The controllers for %s are not registered" % partial_controller)
-
-        return FlowControllerLinear(passes, options)
+        if name not in cls.hierarchy:
+            raise KeyError("Flow controller not found: %s" % name)
+        del cls.registered_controllers[name]
+        cls.hierarchy.remove(name)
 
 
-class FlowControllerLinear(FlowController):
-    """The basic controller runs the passes one after the other."""
+class FlowControllerLiner(BaseFlowController):
+    """A standard flow controller that runs tasks one after the other."""
```

Comment:
(This comment might be too specific for the overview-style review I was trying for, but...) This meant `Linear` not `Liner`.

Reply:
Thanks for catching this. Fixed in [65fc13d](https://github.com/Qiskit/qiskit/pull/10127/commits/65fc13d3c3803696b9ec98b6ba3b78361f159313)